client/src/js/services/grid/GridGrouping.js

```diff
@@ -98,9 +99,32 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     }
   }
 
-  function handleBatchSelection (){
+  function handleBatchSelection() {
     var gridApi = this.gridApi;
+    var gridRows = gridApi.selection.getSelectedGridRows();
+    var parents = {};
+
+    var hasSelections = gridApi.selection.getSelectedRows().length > 0;
+
+    gridRows.forEach(function (row) {
+      var parentRow = row.treeNode.parentRow;
+      if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
+        parentRow.isSelected = true;
+        parents[parentRow.uid] = parentRow;
+        SELECTED_GROUP_HEADERS = parents;
```

Comment:
the `SELECTED_GROUP_HEADERS` is a global variable and must be updated only when there are new parent row (enter in the `if` clause). 

If it is out the `if` we cannot deselect parent rows which were previously selected because 
```javascript
var parents = {};

/* this code is not executed
gridRows.forEach(function (row) {
  var parentRow = row.treeNode.parentRow;
  if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
    parentRow.isSelected = true;
    parents[parentRow.uid] = parentRow;
  }
}); */

SELECTED_GROUP_HEADERS = parents; // SELECTED_GROUP_HEADERS = {}
```